processor.go

```diff
@@ -352,56 +336,50 @@ func (g *Processor) rebalanceLoop(ctx context.Context, consumerGroup sarama.Cons
 		topics = append(topics, g.graph.LoopStream().Topic())
 	}
 
-	var errs = new(multierr.Errors)
+	var errs *multierror.Error
 
 	defer func() {
-		errs.Collect(rerr)
-		rerr = errs.NilOrError()
+		rerr = multierror.Append(errs, rerr).ErrorOrNil()
 	}()
 
 	defer func() {
 		g.log.Debugf("closing consumer group")
 		defer g.log.Debugf("closing consumer group ... done")
```

Comment:
the nested defer and the many defers look like a code-smell (not sure though).
Go will unroll all defers from bottom to top anyway. Do you need the nesting at all?
Maybe you can also group all teardown logs in one defer func?

Reply:
yeah the nesting looks a bit ugly here, but the outer defer is necessary since the loop below never breaks (but returns). But I removed the inner defer to make it look less ugly.